ts/src/core/query/query.ts

```diff
@@ -98,63 +97,33 @@ interface validCursorOptions {
   keys: string[];
 }
 
-function convertToIntMaybe(val: string) {
-  // TODO handle both cases... (time vs not) better
-  // TODO change this to only do the parseInt part if time...
-  // pass flag indicating if time?
-
-  // handle non-integers for which the first part is an int
-  // @ts-ignore
-  if (isNaN(val)) {
-    return val;
-  }
-  const time = parseInt(val, 10);
-  if (isNaN(time)) {
-    return val;
-  }
-  return time;
-}
-
-function assertValidCursor(cursor: string, opts: validCursorOptions): any {
-  let decoded = Buffer.from(cursor, "base64").toString("ascii");
-  let parts = decoded.split(":");
+export type CursorKeyValues = [key: string, value: string | number | null][];
 
+function translateCursorToKeyValues(
+  cursor: string,
+  opts: validCursorOptions,
+): CursorKeyValues {
   const { keys } = opts;
+  const decoded = atob(cursor);
+  let cursorData: CursorKeyValues = [];
+  try {
+    cursorData = JSON.parse(decoded);
+  } catch (error) {
+    throw new Error(`Bad cursor format ${cursor} passed`);
+  }
```

Comment:
can you write something about what changed here? this is a breaking change

this might be the final PR that forces the v0.2.0 and happy yo make it clear

Reply:
You mean as a description for a changelog, right? Not literally in the code.

Reply:
Here's a description:

Cursor values for GraphQL connections have changed, along with how pagination works under the hood. Previously, pagination would only account for a single sort column (not including the primary key for the table). This meant you could paginate a result of users that were sorted by `birthday`, but it would produce incorrect results if you tried to sort by `birthday` and `name`. Additionally, the placement of null values in the result set (first/last) was not preserved between the original fetch and subsequent pagination. Now you can sort by any number of columns and the pagination results will always reference the original sort order, including null placement.

If you were never sorting by more than one column other than the primary key you should not notice any changes. If you were sorting by multiple non-primary key columns, you will notice a more consistent and accurate pagination.

If you were storing graphQL connection cursors to be re-used across requests, you will need to update those cursors or change your implementation to only use the cursors returned from each request.

Reply:
Oh, I should also add another note:

The `dateColumn` property in the `order` field of `EdgeQueryOptions` has been removed. You no longer need to manually specify fields as date types and can remove all reference to this property.